packages/bpk-component-slider/src/BpkSlider.tsx

```diff
@@ -117,20 +126,67 @@ const BpkSlider = ({
 
 // Work around until radix-ui/react-slider is updated to accept an inputRef prop https://github.com/radix-ui/primitives/pull/3033
 const BubbleInput = forwardRef(
-  (props: ComponentPropsWithRef<'input'>, forwardedRef) => {
-    const { value, ...inputProps } = props;
+  (
+    props: ComponentPropsWithRef<'input'> & {
+      thumbRef: RefObject<HTMLElement>;
+    },
+    forwardedRef,
+  ) => {
+    const { thumbRef, value, ...inputProps } = props;
     const ref = useRef<HTMLInputElement>();
     const composedRefs = useComposedRefs(forwardedRef, ref);
 
-    const prevValue = usePrevious(value);
-
-    // Bubble value change to parents (e.g form change event)
+    // Provide native behaviour that the input range type provides.
     useEffect(() => {
-      const input = ref.current!;
-      if (prevValue !== value) {
-        setNativeValue(input, `${value}`);
+      const thumb = thumbRef.current;
+      const input = ref.current;
+      if (thumb) {
+        const interactionEndHandler = (prevValue: number) => {
+          if (input) {
+            const newValue = parseFloat(input.getAttribute('value') as string);
+            if (prevValue !== newValue) {
+              setNativeValue(input, newValue);
+            }
+          }
+        };
+
+        const focusInHandler = () => {
+          // Controller needed as we may click more than once when in focus. On focusout we can then abort the event listeners
+          const controller = new AbortController();
+          thumb.addEventListener('focusout', () => controller.abort(), {
+            once: true,
+          });
+          // needed on document as users can drag the thumb while out of the thumb elements mouse area
+          document.addEventListener(
+            'click',
+            () => interactionEndHandler(value as number),
+            {
+              signal: controller.signal,
+            },
+          );
+          thumb.addEventListener(
+            'keyup',
+            (e: KeyboardEvent) => {
+              ['ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown'].includes(
+                e.key,
+              ) && interactionEndHandler(value as number);
```

Comment:
Should this logic not be within the handler itself? 
Even if there are different types of handlers with shared common logic: 

```js
// common & shared
function interactionValueParsing(){}

function keyupInteractionHandler (e){
   if(!['ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown'].includes( e.key)) return;
   const value = interactionValueParsing(e)
   // stuff
}

Reply:
Well It would mean checking for key events when it was a click event. This just sort of wraps and takes care of the custom keyboard logic.

Reply:
> It would mean checking for key events when it was a click event

I don't understand. Why? Don't we attach the handler twice, once for click and once for keyup - these could just be different handlers, no?